src/earn/EarnPoolInfoScreen.tsx

```diff
@@ -683,6 +765,140 @@ function InfoBottomSheet({
   )
 }
 
+function BeforeDepositBottomSheet({
+  forwardedRef,
+  token,
+  canSameChainSwapToDeposit,
+  canCrossChainSwapToDeposit,
+  canAdd,
+  exchanges,
+}: {
+  forwardedRef: RefObject<BottomSheetModalRefType>
+  token: TokenBalance
+  canSameChainSwapToDeposit: boolean
+  canCrossChainSwapToDeposit: boolean
```

Comment:
This is actually just cross chain swap right? we don't support cross chain swap directly do deposit yet